src/main/java/org/springframework/hateoas/client/Traverson.java

```diff
@@ -344,31 +371,32 @@ public Link asTemplatedLink() {
 		private Link traverseToLink(boolean expandFinalUrl) {
 
 			Assert.isTrue(rels.size() > 0, "At least one rel needs to be provided!");
-			return new Link(traverseToFinalUrl(expandFinalUrl), rels.get(rels.size() - 1));
+			return new Link(traverseToFinalUrl(expandFinalUrl), rels.get(rels.size() - 1).getRel());
 		}
 
 		private String traverseToFinalUrl(boolean expandFinalUrl) {
 
-			String uri = getAndFindLinkWithRel(baseUri.toString(), rels.iterator());
+			String uri = getAndFindLinkWithRel(baseUri.toString(), rels);
 			UriTemplate uriTemplate = new UriTemplate(uri);
 			return expandFinalUrl ? uriTemplate.expand(templateParameters).toString() : uriTemplate.toString();
 		}
 
-		private String getAndFindLinkWithRel(String uri, Iterator<String> rels) {
```

Comment:
Can we stay with the `Iterator` until we really need `List`? It seems we're switching to a different type and then introduce methods to reinstantiate all the traits the original type had: carrying a head and a tail.